common/djangoapps/enrollment/views.py

```diff
@@ -627,7 +627,8 @@ def post(self, request):
                     for attr in enrollment_attributes
                 ]
                 missing_attrs = set(REQUIRED_ATTRIBUTES.get(mode, [])) - set(actual_attrs)
-            if has_api_key_permissions and (mode_changed or active_changed):
+            audit_with_order = mode == 'audit' and 'order:order_number' in actual_attrs
+            if (has_api_key_permissions and (mode_changed or active_changed)) or audit_with_order:
```

Comment:
It feels like this should actually be this:

```python
if has_api_key_permissions and (mode_changed or active_changed or audit_with_order):
```
Are there tests we can update to ensure this works the way we expect it to?

Reply:
i wanted it separately since it is being added for a separate flow as part of this test, i added a comment - since if/when we are "implementing" this it might work differently

i'll try to find a test